### PR TITLE
ALTAPPS-793: Shared streak recovery feature

### PR DESCRIPTION
**Pull request by vladkash**
**YouTrack Issues**:
[#ALTAPPS-793](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-793)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] View analytics events are added for new screens;
- [x] New analytics events are documented;
- [x] All checks have been passed;

**Description**
- Added `StreakRecoveryFeature`
- Implemented embedding `StreakRecoveryFeature` to `AppFeature`

### CODE REVIEW

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/App/ViewControllers/AppViewController.swift [hunk 1]
```diff
@@ -66,6 +66,7 @@ extension AppViewController: AppViewControllerProtocol {
     }
 
     func displayViewAction(_ viewAction: AppFeatureActionViewActionKs) {
+        // TODO: add streak recovery view actions processing
```
Comment (ivan-magda) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/App/ViewControllers/AppViewController.swift:69]: On iOS you could use `#warning("message")`

File: shared/src/commonTest/kotlin/org/hyperskill/app/AppFeatureTest.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package org.hyperskill.app
```
Comment (ivan-magda) [inline shared/src/commonTest/kotlin/org/hyperskill/app/AppFeatureTest.kt:1]: Package name should be `org.hyperskill.main` as in `commonMain`

File: shared/src/commonTest/kotlin/org/hyperskill/app/AppFeatureTest.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        assertTrue {
+            actions.any {
+                it is AppFeature.Action.StreakRecoveryAction &&
+                    it.action is StreakRecoveryFeature.InternalAction.FetchStreak
+            }
+        }
+    }
+
+    @Test
+    fun `Streak recovery should NOT be initialized only when user is NOT authorized`() {
```
Comment (XanderZhu) [inline shared/src/commonTest/kotlin/org/hyperskill/app/AppFeatureTest.kt:30]: Streak recovery should NOT be initialized ~~only~~ when user is NOT authorized

File: shared/src/commonMain/kotlin/org/hyperskill/app/streak_recovery/presentation/StreakRecoveryActionDispatcher.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    config: ActionDispatcherOptions,
+    private val profileInteractor: ProfileInteractor,
+    private val streaksInteractor: StreaksInteractor,
+    private val analyticInteractor: AnalyticInteractor,
+    private val streakFlow: StreakFlow
+) : CoroutineActionDispatcher<Action, Message>(config.createConfig()) {
+    override suspend fun doSuspendableAction(action: Action) {
+        when (action) {
+            StreakRecoveryFeature.InternalAction.FetchStreak -> {
+                // TODO: replace with current profile state repository when ALTAPPS-682 will be ready
```
Comment (ivan-magda) [inline shared/src/commonMain/kotlin/org/hyperskill/app/streak_recovery/presentation/StreakRecoveryActionDispatcher.kt:22]: Now it's possible to use the current profile state repository, needs to merge `develop`

**Review by ivan-magda — APPROVED** [review]

**Comment by vladkash** [comment]
@ivan-magda I added checking feature flag, strings in shared and made some improvements of it's interface

**Comment by ivan-magda** [comment]
> I added checking feature flag, strings in shared and made some improvements of it's interface

cc @XanderZhu ^^

**Pull request merged**